### PR TITLE
Fix parsing and pretty-printing of string literals

### PR DESCRIPTION
**Pull request by Gabriella439**
While debugging the test failure on #218 I realized that the
logic for parsing and pretty-printing import paths was broken in
several ways:

* Import paths were being rendered unquoted (so the rendered .`proto`
  was not valid)

* Import parsing and pretty-printing was not handling escape sequences
  correctly

* The quickcheck generator was not even exercising this code path

  … which is why they remained broken

… so I fixed this

Fixes #218

### CODE REVIEW

File: src/Proto3/Suite/DotProto/Parsing.hs [hunk 1]
```diff
@@ -125,9 +126,71 @@ globalIdentifier = token $ string "." >> _identifier
 ----------------------------------------
 -- values
 
--- [issue] these string parsers are weak to \" and \000 octal codes
-stringLit :: ProtoParser String
-stringLit = stringLiteral <|> stringLiteral'
+strLit :: ProtoParser String
+strLit = doubleQuotedLiteral <|> singleQuotedLiteral
```
Comment (j6carey) [inline src/Proto3/Suite/DotProto/Parsing.hs:130]: To understand the correctness of this parser I would need to find the part of the proto3 specification that specifies the grammar for string literals look like in .proto files, but thus far I have been unable to locate that specification.  Do you know where that documentation resides?

Comment (Gabriella439) [reply src/Proto3/Suite/DotProto/Parsing.hs:130]: You can find the grammar here: https://developers.google.com/protocol-buffers/docs/reference/proto3-spec

The relevant part is:

```abnf
strLit = ( "'" { charValue } "'" ) |  ( '"' { charValue } '"' )
charValue = hexEscape | octEscape | charEscape | /[^\0\n\\]/
hexEscape = '\' ( "x" | "X" ) hexDigit hexDigit
octEscape = '\' octalDigit octalDigit octalDigit
charEscape = '\' ( "a" | "b" | "f" | "n" | "r" | "t" | "v" | '\' | "'" | '"' )
```

Comment (j6carey) [reply src/Proto3/Suite/DotProto/Parsing.hs:130]: Thank you very much!

**Review by j6carey — APPROVED** [review]
Thank you very much for this very thorough fix.

File: src/Proto3/Suite/DotProto/Parsing.hs (hunk 1 above)
Comment (Gabriella439) [reply src/Proto3/Suite/DotProto/Parsing.hs:130]: You're welcome!

**Comment by Gabriella439** [comment]
I had to remove a few tests which were broken.  The root cause was that the proto3 parser was silently ignoring the rest of the file when hitting an invalid parse instead of failing.  This PR fixed that (by requiring the parser to match `eof` when it's done), but then unearthed some invalid proto3 files in the test suite, which I had to delete (along with their reverse dependencies).

**Comment by j6carey** [comment]
Thanks!

**Comment by Gabriella439** [comment]
You're welcome!

**Pull request merged**